docs/Design_for_securing_CLI_properties_with_NodeJS_Crypto.md

```diff
@@ -0,0 +1,310 @@
+
+The retirement of Keytar affects the Zowe CLI's ability to store secure property values. Keytar is being retired as part of the retirement of the Atom text editor (https://github.blog/2022-06-08-sunsetting-atom/). The approach for replacing Keytar is to write a plug-compatible, platform-specific set of library functions in the Rust programming language. The design of that Rust approach is described in a separate design document.
+
+While evaluating alternative solutions to the Keytar retirement, an option to encrypt secure values using NodeJS cryptographic functions was investigated. This document describes an approach that could be used to implement secure storage with portable, NodeJS technology. The purpose of this document is to retain a record of the investigation of this technology should it be valuable for some effort in the future.
+
+## **Advantages of a NodeJS Crypto approach**
+
+### **A NodeJS solution could avoid installation problems** ###
+
+The Keytar component is a platform-specific binary library that is installed during the installation of Zowe CLI.
+
+The Zowe Secure Credential Store (SCS) works reliably after Keytar has been successfully installed. However, customers have encountered problems installing the SCS due to installation problems related to Keytar. Some problems occurred when obtaining the Keytar binary module. Some problems occurred from overlooking the requirement to unlock the Linux keyring. Some problems appear to be due to locked-down security policies at some sites.
+
+Typically, the CLI installation completes and CLI commands will work. However, a CLI command that tries to access data in the SCS fails to load the SCS due to a previously failed SCS installation. The majority of the CLI commands connect to z/OS, which require credentials. Those credentials should be stored in the SCS. Thus, most CLI commands become unusable when the SCS fails to install.
+
+It is felt that an implementation in purely Typescript, using NodeJS-supplied cryptographic functions would not experience the same installation problems.
+
+### **Support for USS** ###
+
+Another scenario for which a NodeJS Crypto solution could provide value is the use of Zowe CLI on USS. Because no Keytar binary is available for USS, customers must currently store credentials in plain text. A Zowe NodeJS Crypto solution will use the same NodeJS-supplied technology on every platform, including USS.
+
+## **Shortcomings of a NodeJS Crypto approach**
+Because a different approach is used for encrypting data, consumers will experience some operational changes after a NodeJS Crypto solution is delivered.
+
+### **Consumers must re-enter all secure values after upgrade** ###
+
+Because secure values will no longer be stored in the same vault, consumers must re-enter their values one time so that they are available for future use by Zowe CLI commands.
```

Comment:
If we adopt the Node.js crypto functions in the future, we could probably use the Rust library to migrate these credentials and avoid re-entry (at least for the operating systems that the Rust implementation is compatible with).

LGTM!